pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1202,6 +1211,22 @@ type Sigv4 struct {
 	RoleArn string `json:"roleArn,omitempty"`
 }
 
+// AzureAD defines the configuration for remote write's azuread parameters.
+// +k8s:openapi-gen=true
+type AzureAD struct {
+	// The Azure Cloud. Options are 'AzurePublic', 'AzureChina', or 'AzureGovernment'. If blank, AzurePublic is used.
+	Cloud string `json:"cloud,omitempty"`
+	// ManagedIdentity defines the Azure User-assigned Managed identity.
+	ManagedIdentity *ManagedIdentity `json:"managedIdentity"`
```

Comment:
If it might be null worth to add `// +optional` annotation

Reply:
It seems to me that ClientID is a mandatory field, hence neither  `managedIdentity` nor `managedIdentity.clientID` can't be optional.

https://github.com/prometheus/prometheus/blob/1b9a53b5eea62e82a65bb65a8139c38dbd8f259e/storage/remote/azuread/azuread.go#L87-L93

Reply:
yup the `managedIdentity` part is intended 😄

Reply:
Could we add the comment `// +required` and remove the pointer then?

Reply:
added `// +required` and removed pointer.